### PR TITLE
Removes unused files, types, exports, dupe exports, and adds unlisted deps

### LINKED ISSUE #47971: Clean up unused files and exports
As discussed in https://github.com/freeCodeCamp/freeCodeCamp/pull/47836 there are several unused files and exports.  We should remove all of these from the codebase.

The full set of tasks is:

- [ ] Remove unused files
- [ ] Add missing dependencies
- [ ] Remove unused exports
- [ ] Remove unused types
- [ ] Remove duplicate exports

The files with issues are:

--- UNUSED FILES (8)
src/assets/icons/donate-with-PayPal.tsx
src/assets/icons/intro-information.tsx
src/components/Donation/duration-amount-options.tsx
src/components/Header/components/user-state.tsx
src/components/helpers/form/block-save-wrapper.tsx
src/components/search/searchPage/empty-search.tsx
src/components/search/searchPage/no-results.tsx
src/components/search/searchPage/search-page-hits.tsx

--- UNLISTED DEPENDENCIES (4)
react-instantsearch-core  src/components/search/searchBar/search-bar.tsx
react-instantsearch-core  src/components/search/searchBar/search-hits.tsx
react-instantsearch-core  src/components/search/searchBar/search-suggestion.tsx
gatsby-link               src/templates/Introduction/components/cert-challenge.tsx

--- UNUSED EXPORTS (46)
DonationOptionsText              src/components/Donation/donation-text-components.tsx
DonationSupportText              src/components/Donation/donation-text-components.tsx
PaypalButton                     src/components/Donation/paypal-button.tsx
sagas                            src/components/Flash/redux/index.ts
NavLinks                         src/components/Header/components/nav-links.tsx
searchIndexNameSelector          src/components/search/redux/index.js
updateSearchIndexName            src/components/search/redux/index.js
SearchBarOptimized               src/components/search/searchBar/search-bar-optimized.tsx
SearchBar                        src/components/search/searchBar/search-bar.tsx
CertificationSettings            src/components/settings/certification.js
updateCurrentChallengeId         src/redux/actions.js
initialState                     src/redux/index.js
defaultFetchState                src/redux/index.js
CurrentCertsPropType             src/redux/prop-types.ts
UserPropType                     src/redux/prop-types.ts
saveChallengeSaga                src/redux/save-challenge-saga.js
sessionMetaSelector              src/redux/selectors.js
defaultDonationFormState         src/redux/types.ts
defaultFetchState                src/redux/types.ts
CompletionModalInner             src/templates/Challenges/components/completion-modal.tsx
getCompletedPercent              src/templates/Challenges/components/completion-modal.tsx
HelpModal                        src/templates/Challenges/components/help-modal.tsx
ShortcutsModal                   src/templates/Challenges/components/shortcuts-modal.tsx
VideoModal                       src/templates/Challenges/components/video-modal.tsx
ToolPanel                        src/templates/Challenges/projects/tool-panel.tsx
replaceNBSP                      src/templates/Challenges/rechallenge/transformers.js
testJS$JSX                       src/templates/Challenges/rechallenge/transformers.js
checkChallenge                   src/templates/Challenges/redux/actions.js
lockCode                         src/templates/Challenges/redux/actions.js
updateJSEnabled                  src/templates/Challenges/redux/actions.js
updateSuccessMessageSaga         src/templates/Challenges/redux/current-challenge-saga.js
currentChallengeSaga             src/templates/Challenges/redux/current-challenge-saga.js
executeChallengeSaga             src/templates/Challenges/redux/execute-challenge-saga.js
executeCancellablePreviewSaga    src/templates/Challenges/redux/execute-challenge-saga.js
executeCancellableChallengeSaga  src/templates/Challenges/redux/execute-challenge-saga.js
currentTabSelector               src/templates/Challenges/redux/selectors.js
buildBackendChallenge            src/templates/Challenges/utils/build.ts
buildJSChallenge                 src/templates/Challenges/utils/build.ts
buildDOMChallenge                src/templates/Challenges/utils/build.ts
Block                            src/templates/Introduction/components/block.tsx
CertChallenge                    src/templates/Introduction/components/cert-challenge.tsx
handledErrorSymbol               src/utils/handled-error.ts
wrapHandledError                 src/utils/handled-error.ts
pathParsers                      src/utils/path-parsers.ts
pathsMap                         utils/challenge-types.js
injectConditionalTags            utils/tags.tsx

--- UNUSED TYPES (17)
LoginProps                interface  src/components/Header/components/Login.tsx
AuthOrProfileProps        interface  src/components/Header/components/auth-or-profile.tsx
MenuButtonProps           interface  src/components/Header/components/menu-button.tsx
NavLinksProps             interface  src/components/Header/components/nav-links.tsx
UniversalNavProps         interface  src/components/Header/components/universal-nav.tsx
HeaderProps               interface  src/components/Header/index.tsx
Validator                 type       src/components/formHelpers/form-validators.tsx
FormProps                 type       src/components/formHelpers/form.tsx
FileKeyChallenge          type       src/redux/prop-types.ts
SavedChallenges           type       src/redux/prop-types.ts
ClaimedCertifications     type       src/redux/prop-types.ts
CertTest                  type       src/redux/prop-types.ts
AllMarkdownRemark         type       src/redux/prop-types.ts
Required                  type       src/redux/prop-types.ts
DefaultDonationFormState  interface  src/redux/types.ts
DefaultFetchState         interface  src/redux/types.ts
HandledError              type       src/utils/handled-error.ts

--- DUPLICATE EXPORTS (4)
PayPalButtonScriptLoader, default src/components/Donation/paypal-button-script-loader.tsx
UniversalNav, default             src/components/Header/components/universal-nav.tsx
Header, default                   src/components/Header/index.tsx
Map, default                      src/components/Map/index.tsx

### PR DESCRIPTION
**Pull request by tom-raley**
Checklist:

- [X] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Closes #47971

* Should close all checklist items from #47971 in a single PR. I think I covered everything, but not 100% sure.
* I had to leave some functions/classes as exports because they were actually being used and needed to be tested in unit tests. This was only the case with a few of them.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -105,7 +105,9 @@
   },
   "dependencies": {
     "dotenv": "10.0.0",
-    "invariant": "2.2.4"
+    "gatsby-link": "^4.24.1",
+    "invariant": "2.2.4",
```
[the remark this answers is not in the document]
Comment (tom-raley) [reply package.json:109]: I thought that's what the list meant by "Unlisted Dependencies", that these are being used but they're not listed as dependencies in the package.json. I see it's causing issues and the project hardly has any dependencies so I'm guessing that's wrong ☺️

**Comment by ojeytonwilliams** [comment]
Hi @tom-raley thanks for your hard work on this, it's looking great.

We just merged a couple of PRs that dealt with some of the tasks. I believe there are some minor conflicts that GitHub has not picked up on.  If you could rebase your PR onto main, that would be awesome.

Finally, a few tests have started failing. Could you see what's going on there?

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (ShaunSHamilton) [reply package.json:109]: To be clear, we use [NPM workspaces](https://docs.npmjs.com/cli/v8/using-npm/workspaces), so dependencies should be updated/included as such.

**Comment by raisedadead** [comment]
Hi, 

We are closing old issues/pull requests that we think are stale. If you still need help or are fixing this, please open a new issue/pull request.

Thanks for your understanding. 

Happy contributing.

**Pull request closed without merging**